clients/iceberg-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSqlExtensionsAstBuilder.scala

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst.parser.extensions
+
+import org.antlr.v4.runtime._
+import org.antlr.v4.runtime.misc.Interval
+import org.antlr.v4.runtime.tree.{ParseTree, TerminalNode}
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal}
+import org.apache.spark.sql.catalyst.parser.ParserInterface
+import org.apache.spark.sql.catalyst.parser.extensions.NessieParserUtils.withOrigin
+import org.apache.spark.sql.catalyst.parser.extensions.NessieSqlExtensionsParser._
+import org.apache.spark.sql.catalyst.plans.logical.{
+  CallArgument,
+  CreateReferenceField,
+  DropReferenceField,
+  ListReferenceField,
+  LogicalPlan,
+  MergeBranchField,
+  NamedArgument,
+  PositionalArgument,
+  ShowLogField,
+  ShowReferenceField,
+  UseReferenceField
+}
+import org.apache.spark.sql.catalyst.trees.{CurrentOrigin, Origin}
+
+import scala.collection.JavaConverters._
+
+class NessieSqlExtensionsAstBuilder(delegate: ParserInterface)
+    extends NessieSqlExtensionsBaseVisitor[AnyRef] {
+
+  override def visitNessieCreateRef(
+      ctx: NessieCreateRefContext
+  ): CreateReferenceField = withOrigin(ctx) {
+    val isBranch = ctx.TAG == null
+    val refName = ctx.identifier(0).getText
+    val catalogName = Option(ctx.catalog).map(x => x.getText)
```

Comment:
Can you add a utility method to simplify those `Option(ctx.foobar).map(x => x.getText)` to something like `asText(ctx.foobar)`.

Reply:
fixed